pkg/ottl/compare.go

```diff
@@ -135,6 +136,48 @@ func (p *Parser[K]) compareFloat64(a float64, b any, op compareOp) bool {
 	}
 }
 
+func (p *Parser[K]) compareTime(a time.Time, b any, op compareOp) bool {
+	switch v := b.(type) {
+	case time.Time:
+		switch op {
+		case EQ:
+			if a.Equal(v) == true {
+				return true
+			}
+			return false
```

Comment:
```suggestion
			return a.Equal(v)
```

Let's simplify the different operation cases to return the value returned by the function.